src/ui/components/ProductList.tsx

```diff
@@ -0,0 +1,21 @@
+import { ProductElement } from "./ProductElement";
+import { type ProductFragment } from "@/gql/graphql";
+
+type Props = {
+	products: ProductFragment[];
+};
+
+export const ProductList = ({ products }: Props) => {
+	return (
+		<ul className="grid gap-8 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4" role="list">
+			{products.map((product, index) => (
+				<ProductElement
+					key={product.id}
+					product={product}
+					loading={index < 4 ? "eager" : "lazy"}
+					sizes="(max-width: 640px) 100vw, (max-width: 768px) 50vw, (max-width: 1024px) 33vw, 25vw"
```

Comment:
Where do these values come from?
What are we trying to achieve here, given that the performance score in Lighthouse is already 100/100?

Reply:
@mmiszy Great questions!

The values referenced in the media queries in sizes attribute are derived from Tailwind CSS breakpoints, serving as the breakpoints used by it - and I know It looks little messy, but I couldn't found better way to provide them here in this case. The second argument in these media queries depends on the grid template applied to the parent ul element - for instance, `grid-cols-3` => `33vw`.

But I why even add this it here? What I want achive? As you mentioned, the application achieves a high Lighthouse score, often reaching 100/100. However, there is recurring feedback indicating that image sizes could be better adjusted, particularly during mobile device testing.

![pagespeed web dev_analysis_https-storefront-saleor-example-vercel-app_vi1ifnwc8x_form_factor=mobile](https://github.com/saleor/storefront/assets/27455716/d222d4a9-20fd-4bf2-a338-d8dcafaa4bea)

I believe that providing the sizes attribute will contribute positively to this metric. While the `srcset` attribute, automatically provided by next/image, allows the browser to decide which image to load based on pixel density, the `sizes` attribute offers additional information about the real space an image will occupy.

I might be wrong. Maybe it is unnecessary and present results are good enough.

Reply:
Okay, but grid-cols-3 is not exactly 33vw, right?

Reply:
> Okay, but grid-cols-3 is not exactly 33vw, right?

Yeah, that's true. I think it isn't possible to provide a exact value of space which image will occupy since we can only pass a value in `vw` units. I tested results of this PR in Lighthouse and it still notify about possibility to shrink images. despite this in the end it reduced amount of data required to load page. Maybe it's not worth sacrificing code readability 🤔